lib/queue.rb

```diff
@@ -1,31 +1,59 @@
 class Queue
-
+  
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(10)
+    @front = -1
+    @back = -1
   end
-
+  
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      front = 0
+      back = 0
+    end
+    if @front == @back
+      #look at test otherwise WE decide
+    end
+    @store[@back] = element
+    @back = (@back +1) % @store.length
   end
-
+  
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if @front == @back 
+      raise ArgumentError.new("Danger, Danger!")
+    end
+    
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = -1
+    end
```

Comment:
Wait!  If the Queue is empty (both front and back are negative 1), Then you just want to raise an error instead of messing with the front and back.